python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -179,6 +211,57 @@ def getURI(
         # Docstring inherited.
         raise NotImplementedError()
 
+    def get_dataset_type(self, name: str) -> DatasetType:
+        # In future implementation this should directly access the cache
+        # and only go to the server if the dataset type is not known.
+        path = f"dataset_type/{name}"
+        response = self._client.get(self._get_url(path))
+        response.raise_for_status()
+        return DatasetType.from_simple(SerializedDatasetType(**response.json()), universe=self.dimensions)
+
+    def get_dataset(self, id: DatasetId) -> DatasetRef | None:
+        path = f"dataset/{id}"
+        response = self._client.get(self._get_url(path))
+        if response.json() is None:
```

Comment:
Does this need some error handling before you try to covert response to JSON (`raise_for_status()`?)

Reply:
Yes. Oops.

Reply:
Turns out FastAPI exception handling is an adventure (you have to raise an HttpException and fill it with all the real exception information and then re-raise in the client). I think I'm going to punt on this for now.